### DISCUSSION TITLE
Rabbit MQ 4.0 release - Will this plugin be supported or work?

### DISCUSSION THREAD

**Discussion by martywaz** [rabbitmq/rabbitmq-delayed-message-exchange#286] [General]
I heard 4.0 is moving to a different database or database model for storage.  Will the Rabbit delayed message exchange be modified and supported for this?

**Comment by michaelklishin** [comment]
The original schema data store will still be around for plugins that need it. Our team works on a commercial alternative to this plugin, so our time investment into this plugin and its current (full of documented limitations) design will be minimal.

That said, since Mnesia will still be on the code path, plugins such as this can start it and use it just like before. Not to mention the fact that Khepri will not be enabled by default in 4.0 (it likely will in 4.1, time will tell).

**Comment by martywaz** [comment]
Thanks.  Can you point me to the commercial alternative to this plugin that you mentioned?

**Reply by michaelklishin** [reply]
It is not ready right now but one day (I cannot make any forward looking statements) it will ship as part of [Tanzu RabbitMQ](https://tanzu.vmware.com/rabbitmq), the commercial edition by Broadcom.
